src/main/java/dev/morling/kccli/util/ConfigurationContext.java

```diff
@@ -15,40 +15,69 @@
  */
 package dev.morling.kccli.util;
 
-import java.io.IOException;
-import java.net.URI;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.StandardOpenOption;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import dev.morling.kccli.service.Configuration;
+import dev.morling.kccli.service.Context;
 
 import javax.enterprise.context.ApplicationScoped;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
 
 @ApplicationScoped
 public class ConfigurationContext {
-
     private static final String CONFIG_FILE = ".kcctl";
-    private static final Path CONFIG_PATH = Path.of(System.getProperty("user.home"), CONFIG_FILE);
+    private final File configPath = new File(System.getProperty("user.home"), CONFIG_FILE);
+    private final ObjectMapper objectMapper;
+
+    public ConfigurationContext() {
+        this.objectMapper =
+            JsonMapper
+                .builder()
+                .serializationInclusion(JsonInclude.Include.NON_NULL)
+                .configure(SerializationFeature.INDENT_OUTPUT, true)
+                .build();
+    }
 
-    public void setConfiguration(String cluster) {
+    public void setContext(String contextName, Context context) {
         try {
-            Files.writeString(CONFIG_PATH, "cluster=" + cluster, StandardOpenOption.TRUNCATE_EXISTING);
-        }
-        catch (IOException e) {
-            throw new RuntimeException("Couldn't write configuration file ~/" + CONFIG_FILE, e);
+            if (!configPath.exists()) {
+                objectMapper.writeValue(
+                    configPath,
+                    new Configuration(contextName).addConfigurationContext(contextName, context)
+                );
+
+                return;
+            }
+
+            var configuration = objectMapper.readValue(configPath, Configuration.class);
+
+            objectMapper.writeValue(
+                configPath,
+                configuration.addConfigurationContext(contextName, context)
+            );
+        } catch (IOException e) {
+            throw new RuntimeException("Couldn't write context file ~/" + CONFIG_FILE, e);
         }
     }
 
     public URI getCluster() {
```

Comment:
Should this method go away entirely, and all access be via `getContext()`? And we'd log an error if no context has been defined.

Reply:
By logging you mean logging to STDOUT?

Reply:
Yes, exactly.